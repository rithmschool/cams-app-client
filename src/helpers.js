export const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'

export const ensureCorrectUser = (pathId) => {
  let token = localStorage.getItem('token');
  if(!token){return false;}
  let tokenId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
  return +tokenId === +pathId;
}

export const userID = () => JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

export const config = {
  headers: {
    'Accept':'application/json',
    'ContentType':'application/json',
    'Authorization':'bearer ' + localStorage.getItem('token')
  }
}
