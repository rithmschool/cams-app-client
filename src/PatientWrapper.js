import React, { Component } from "react";
import PatientHome from "./PatientHome";
import { BASE_URL, BrowserDetect } from "./helpers.js";
import axios from "axios";
import URLSearchParams from "url-search-params";

class PatientWrapper extends Component {
  constructor(props) {
    super(props);

    this.state = {
      screens: [],
      assessmentId: 0,
      token: "",
      src: null,
      screenCount: 0
    };
  }

  captureUserMedia(callback) {
    let params = { audio: true, video: true };
    navigator.getUserMedia(params, callback, error => {
      alert(JSON.stringify(error));
    });
  }

  componentWillMount() {
    const query = new URLSearchParams(this.props.location.search);
    const token = query.get("token");
    console.log(token);

    axios
      .get(`${BASE_URL}/api/users/confirm/${token}`)
      .then(response => {
        this.setState({ assessmentId: response.data.assessment_id });
        console.log(`${BASE_URL}/api/users/confirm/${token}`);
        return axios.get(
          `${BASE_URL}/api/users/${response.data
            .doctor_id}/assessments/${response.data.assessment_id}`,
          { token: token }
        );
      })
      .then(response => {
        this.setState({
          screens: response.data.screens,
          screenCount: response.data.screens.reduce(
            (prev, curScreen) => prev + (curScreen.type === "video" ? 3 : 2),
            3
          )
        });
      });
  }

  render() {
    BrowserDetect.init();
    let display =
      BrowserDetect.browser === "Chrome" ? (
        <PatientHome
          screens={this.state.screens}
          screenCount={this.state.screenCount}
          assessmentId={+this.state.assessmentId}
        />
      ) : (
        <p>
          Browser not supported. Please switch to{" "}
          <a href="https://www.google.com/chrome/browser/desktop/index.html">
            Google Chrome
          </a>{" "}
          to proceed.
        </p>
      );
    return <div>{display}</div>;
  }
}

export default PatientWrapper;
