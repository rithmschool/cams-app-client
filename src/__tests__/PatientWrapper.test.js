import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import PatientWrapper from "../PatientWrapper";
import toJson from "enzyme-to-json";
import LocalStorageMock from "../setupTests/LocalStorageMock";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import { BASE_URL } from "../helpers";

var mock = new MockAdapter(axios);

beforeAll(() => {
  localStorage.setItem("token", "");
  mock
    .onGet(`${BASE_URL}/api/users/confirm/test`)
    .reply(function(config) {
      return [200, { assessment_id: 1, doctor_id: 1 }];
    })
    .onGet(`${BASE_URL}/api/users/1/assessments/1`)
    .reply(200, {
      screens: [
        {
          duration: "PT4S",
          id: 4,
          timer: 30,
          title: "Short funny video",
          type: "video",
          url: "https://www.youtube.com/watch?v=B7bqAsxee4I",
          youtube_id: "B7bqAsxee4I"
        },
        {
          duration: null,
          id: 10,
          timer: 5,
          title: "how are you?",
          type: "question",
          url: null,
          youtube_id: null
        }
      ]
    });
});

test("shapshot of an invalid browser", () => {
  const patientWrapper = shallow(
    <PatientWrapper location={{ search: "?token=test" }} />
  );
  patientWrapper.setState({
    screens: [],
    assessmentId: 0,
    token: "",
    src: null,
    screenCount: 0
  });
  expect(toJson(patientWrapper)).toMatchSnapshot();
});

afterAll(() => {
  mock.restore();
});
