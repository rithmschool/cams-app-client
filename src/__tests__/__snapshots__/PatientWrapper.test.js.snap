// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shapshot of an invalid browser 1`] = `
<div>
  <p>
    Browser not supported. Please switch to
     
    <a
      href="https://www.google.com/chrome/browser/desktop/index.html"
    >
      Google Chrome
    </a>
     
    to proceed.
  </p>
</div>
`;
